graphql/src/main/java/com/linecorp/armeria/server/graphql/DefaultGraphqlService.java

```diff
@@ -100,41 +97,9 @@ private HttpResponse execute(ServiceRequestContext ctx, ExecutionInput input, Me
         } else {
             future = graphQL.executeAsync(input);
         }
-        return HttpResponse.from(future.handle((executionResult, cause) -> {
-            if (cause != null) {
-                // graphQL.executeAsync() returns an error in the executionResult with getErrors().
-                // Use 500 Internal Server Error because this cause might be unexpected.
-                final ExecutionResult error = newExecutionResult(cause);
-                return HttpResponse.ofJson(HttpStatus.INTERNAL_SERVER_ERROR, produceType,
-                                           error.toSpecification());
-            }
-            final List<GraphQLError> errors = executionResult.getErrors();
-            if (!errors.isEmpty() && errors.stream().anyMatch(ValidationError.class::isInstance)) {
-                // The server SHOULD deny execution with a status code of 400 Bad Request for
-                // invalidate documentation.
-                return HttpResponse.ofJson(HttpStatus.BAD_REQUEST, produceType,
-                                           executionResult.toSpecification());
-            }
-            return toHttpResponse(executionResult, produceType);
-        }));
-    }
-
-    private static HttpResponse toHttpResponse(ExecutionResult executionResult, MediaType produceType) {
-        // TODO: When WebSocket is implemented, it should be removed.
-        if (executionResult.getData() instanceof Publisher) {
-            logger.warn("executionResult.getData() returns a {} that is not supported yet.",
-                        executionResult.getData().toString());
-            final ExecutionResult error =
-                    newExecutionResult(new UnsupportedOperationException("WebSocket is not implemented"));
-            return HttpResponse.ofJson(HttpStatus.NOT_IMPLEMENTED, produceType, error.toSpecification());
-        }
-        return HttpResponse.ofJson(produceType, executionResult.toSpecification());
-    }
 
-    private static ExecutionResult newExecutionResult(Throwable cause) {
-        return new ExecutionResultImpl(GraphqlErrorException.newErrorException()
-                                                            .message(cause.getMessage())
-                                                            .cause(cause)
-                                                            .build());
+        return HttpResponse.from(
+                future.handle((executionResult, cause) ->
+                                      errorsHandler.handle(ctx, input, produceType, executionResult, cause)));
```

Comment:
This is no longer an `errorHandler` because it produces a response even if the result is successful.
How about calling `handle` only if there's an error?
```java
return HttpResponse.from(
        future.handle((executionResult, cause) -> {
            if (cause == null && executionResult.getErrors().isEmpty()) {
                return HttpResponse.ofJson(produceType, executionResult.toSpecification());
            }
            if (executionResult.getData() instanceof Publisher) {
                logger.warn("executionResult.getData() returns a {} that is not supported yet.",
                            executionResult.getData().toString());
                final ExecutionResult error = newExecutionResult(
                        new UnsupportedOperationException("WebSocket is not implemented"));
                return HttpResponse.ofJson(HttpStatus.NOT_IMPLEMENTED, produceType,
                                           error.toSpecification());
            }

            return errorsHandler.handle(ctx, input, produceType, executionResult, cause);
        }));
```